src/graphql/resolvers/monsterResolver.ts

```diff
@@ -45,6 +46,29 @@ const resolveDamage = async (damage: Damage[]) => {
 };
 
 const Monster = {
+  armor_class: async (monster: Monster) => {
+    const resolvedAC = monster.armor_class.map(async ac => {
+      const newAC: Record<string, any> = { ...ac };
+
+      if (ac.type === 'armor') {
+        newAC.armor = ac.armor?.map(async armor => {
+          return await EquipmentModel.findOne({ index: armor.index }).lean();
+        });
+      }
```

Comment:
Instead of doing a db request for each piece of armor, you can do it in one request instead if you do the following:
```typescript
 newAC.armor = await EquipmentModel.find({ index: { $in: ac.armor?.map(({ index }) => index) ?? [] } });
```

Reply:
Good idea!